inference/sources/alchemer/main.go

```diff
@@ -0,0 +1,337 @@
+	e := json.Unmarshal(b, &vv)
+
+	// Alchemer sends url_variables as an empty array when empty
+	// this will then return an error when unmarshalling, and
+	// we create an empty variables
+	if e != nil {
+		*v = URLVariables{}
+		return nil
+	}
+	*v = vv
+	return nil
+}
+
+type SurveyDatum struct {
+	Options map[string]struct {
+		ID             int    `json:"id"`
+		Option         string `json:"option"`
+		OriginalAnswer string `json:"original_answer"`
+		Answer         string `json:"answer"`
+	} `json:"options"`
+	ID             int    `json:"id"`
+	Type           string `json:"type"`
+	Question       string `json:"question"`
+	SectionID      int    `json:"section_id"`
+	OriginalAnswer string `json:"original_answer"`
+	Answer         string `json:"answer"`
+	AnswerID       int    `json:"answer_id"`
+	Shown          bool   `json:"shown"`
+}
+
+type GetResponsesResponse struct {
+	ResultOk   bool `json:"result_ok"`
+	TotalCount int  `json:"total_count"`
+	Page       int  `json:"page"`
+	TotalPages int  `json:"total_pages"`
+
+	// can be string or int...
+	// ResultsPerPage string  `json:"results_per_page"`
+	Data []struct {
+		ID            string                     `json:"id"`
+		ContactID     string                     `json:"contact_id"`
+		Status        string                     `json:"status"`
+		IsTestData    string                     `json:"is_test_data"`
+		DateSubmitted string                     `json:"date_submitted"`
+		SessionID     string                     `json:"session_id"`
+		Language      string                     `json:"language"`
+		DateStarted   string                     `json:"date_started"`
+		LinkID        string                     `json:"link_id"`
+		URLVariables  URLVariables               `json:"url_variables,omitempty"`
+		IPAddress     string                     `json:"ip_address"`
+		Referer       string                     `json:"referer"`
+		UserAgent     string                     `json:"user_agent"`
+		ResponseTime  int                        `json:"response_time"`
+		Longitude     string                     `json:"longitude"`
+		Latitude      string                     `json:"latitude"`
+		Country       string                     `json:"country"`
+		City          string                     `json:"city"`
+		Region        string                     `json:"region"`
+		Postal        string                     `json:"postal"`
+		Dma           string                     `json:"dma"`
+		SurveyData    map[string]json.RawMessage `json:"survey_data,omitempty"`
+	} `json:"data"`
+}
+
+type AlchemerError struct {
+	ResultOk bool   `json:"result_ok"`
+	Message  string `json:"message"`
+	Code     string `json:"code"`
+}
+
+func (e *AlchemerError) Empty() bool {
+	return e.Code == ""
+}
+
+func (e *AlchemerError) Error() string {
+	return e.Message
+}
+
+func (c *AlchemerConnector) loadEnv() *AlchemerConnector {
+	err := env.Parse(c)
+	handle(err)
+	return c
+}
+
+type AlchemerCreds struct {
```

Comment:
Could we put this in its own package as the api backend will need to use it for the connected accounts

Reply:
I moved around Flys one into a package called `types`